03-task-manager/starter/middleware/error-handler.js

```diff
@@ -0,0 +1,13 @@
+const { createCustomError, CustomAPIError } = require("../errors/custom-error");
+
+const errorHandlerMiddleware = (err, req, res, next) => {
+  if (err instanceof CustomAPIError) {
+    return res.status(err.statusCode).json({ msg: err.message });
+  }
+  //   console.log(err);
+  //   return res.status(err.status).json({ msg: err.message });
+  //   return res.status(500).json({ msg: err });
+  return res.status(500).json({ msg: "Something went wrong, try again later" });
```

Comment:
Add the err variable to the JSON so the user can know what condition failed in the validation, for example, when the name is not provided to create the task.
You can do it like this:
`return res.status(500).json({ msg: "Something went wrong, try again later", err});`

Reply:
Oh! Great idea! During React practice I struggled because I didn't know what was really wrong (we had backend code without this additional error variable)! Thank you!